### ISSUE TITLE
The frame rate of large models is lower than before

### ISSUE THREAD

**Issue by wwjje** [vimaec/vim-web#7]
The frame rate of large models is lower than before. The previous version had 60 frames per second, but now it only has a frame rate of 20-30 frames per second

**Comment by wwjje** [comment]
May I ask when this issue can be resolved? The performance and frame rate are not as smooth as before when using large models

**Comment by mavimaec** [comment]
Hi wwjje, @vim-sroberge is currently working on a fix. Hoping to have it merged in the next few days.

**Comment by vim-sroberge** [comment]
Hi wwjje, The new release 0.39.0 includes performance improvements.
If you need maximum performance you can also consider using a material override like this

`vim.scene.material = viewer.materials.simple`

**Comment by wwjje** [comment]
Hi，I tried it out，The frame rate has increased to around 40FPS, but still hasn't reached 60FPS. During use, there may be a slight lag and lack of smoothness when zooming in, out, or rotating. I tried another version (https://github.com/vimaec/vim-webgl-viewer/tree/v1.5.2) again, and no matter how I operate it, zoom in, zoom out, or rotate, it always has a full frame rate of 60FPS. Why is that?

**Comment by wwjje** [comment]
I used the same computer and the same large model for testing. You can also test it. vim-webgl-viewer v1.5.2 performs very smoothly, always running at a full frame rate of 60FPS

**Comment by vim-sroberge** [comment]
I will look into it and try an older version. Thanks for the details.

**Comment by wwjje** [comment]
May I ask what the reason is? Is it easy to fix?

**Comment by vim-sroberge** [comment]
You can greatly improve performance by using FrontSide rendering only instead of Two-Sided.
Especially the standard transparent material.

It can cause visual artifacts for some geometry.
It causes missing surfaces when using the section box as well.
But it is much faster.

You can change it at run time with
`viewer.materials.transparent.material.side = 0`

There are now plans to add this option to the settings UI.

**Comment by wwjje** [comment]
Yes, I did make the modifications much faster. Will the impact be significant after the modifications? What is VIM Ultra? I can't run it to check, it seems to be missing the address of the server's websocket

**Comment by mavimaec** [comment]
@wwjje VIM Utra is part of the VIM Desktop application. It allows you to stream our Vulkan-based renderer from your local machine into your web browser. Here is our documentation for [VIM Ultra](https://docs.vimaec.com/docs/vim-for-windows/configuring-vim-ultra). We use it with our Power BI custom visual to render the largest BIM models.

**Comment by wwjje** [comment]
Do I need to use Vim.Enterprise? It requires a license, is there a trial version available?

**Comment by mavimaec** [comment]
You don't need to use VIM Enterprise to run VIM Ultra. All you need is the VIM Desktop app which is available in the standard "VIM Desktop Installer" (see our [release notes for download links](https://docs.vimaec.com/docs/release-notes/release-notes))

**Comment by wwjje** [comment]
I am running it now, but it seems that I can only see the example model. If I change the local model or other models, WebGL can view it, but Ultra cannot

**Comment by vim-sroberge** [comment]
You should be able to change the ultra demo code to point to any file you want, the file can either be local or remote. The server will then download the file and display it.

**Comment by wwjje** [comment]
I changed the addresses to‘ http://192.168.1.3:5174/vim-web/new.vim ','./new.vim 'and the download url and webgl that can be accessed by the Internet can be displayed normally, and Ultra prompts download error,But the Vim example can be opened normally.





I also need to add the following configurations in order to run properly, otherwise an error will occur. I'm not sure if it will have any impact

**Comment by wwjje** [comment]
![Image](https://github.com/user-attachments/assets/06ffb8b6-709f-42e2-a18f-e3fef30b3309)

![Image](https://github.com/user-attachments/assets/21a3bbe6-bc8d-4da4-a218-5d9175856f83)

![Image](https://github.com/user-attachments/assets/cb16c0eb-9768-4b8a-a108-04f52fed427d)

![Image](https://github.com/user-attachments/assets/e9abc019-65a7-4cdc-ac13-dc95214b0a38)

**Comment by mavimaec** [comment]
@wwjje for VIM Ultra to work, the VIM file needs to be hosted from a server which can respond to HTTP GET requests.

**Comment by vim-sroberge** [comment]
You can load local files, but they need to be put inside the repo so you can reference them with a relative path. 
Then you could change `src/pages/ultra/00_home.tsx` to load that file instead. 

That is how I usually test locally.

`  const url = getPathFromUrl() ?? Urls.residence //Not used in this example
  const request = webgl.loader.request(
    { url:'./residence.vim' }
  )
`
